shared/fs/common/hooks.tsx

```diff
@@ -60,12 +60,12 @@ export const useFsPathMetadata = (path: Types.Path) => {
   }, [dispatch, path])
 }
 
-export const useFsChildren = (path: Types.Path) => {
+export const useFsChildren = (path: Types.Path, initialLoadRecursive?: boolean) => {
   useFsPathSubscriptionEffect(path, RPCTypes.PathSubscriptionTopic.children)
   const dispatch = useDispatchWhenConnectedAndOnline()
   React.useEffect(() => {
-    isPathItem(path) && dispatch(FsGen.createFolderListLoad({path}))
-  }, [dispatch, path])
+    isPathItem(path) && dispatch(FsGen.createFolderListLoad({path, recursive: !!initialLoadRecursive}))
```

Comment:
```suggestion
    isPathItem(path) && dispatch(FsGen.createFolderListLoad({path, recursive: initialLoadRecursive || false}))
```
This is more clearly coercing an undefined, in my eyes. Open to disagreement though.